apps/astarte_appengine_api/test/support/generators/interface-value.ex

```diff
@@ -0,0 +1,128 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.AppEngine.API.InterfaceValueTestGenerator do
+  use ExUnitProperties
+
+  def gen_datetime() do
+    StreamData.fixed_map(%{
+      year: StreamData.integer(2020..2030),
+      month: StreamData.integer(1..12),
+      day: StreamData.integer(1..28),
+      hour: StreamData.integer(0..23),
+      minute: StreamData.integer(0..59),
+      second: StreamData.integer(0..59),
+      millisecond: StreamData.integer(0..999)
+    })
+    |> StreamData.map(fn %{
+                           year: y,
+                           month: mo,
+                           day: d,
+                           hour: h,
+                           minute: mi,
+                           second: s,
+                           millisecond: ms
+                         } ->
+      {:ok, date} = Date.new(y, mo, d)
+      {:ok, time} = Time.new(h, mi, s, {ms, 3})
+      {:ok, datetime} = DateTime.new(date, time, "Etc/UTC")
+      DateTime.to_iso8601(datetime)
+    end)
+  end
+
+  def gen_datetime_array() do
+    StreamData.list_of(gen_datetime(), min_length: 1, max_length: 10)
+  end
+
+  def gen_invalid_datetime() do
+    StreamData.string(:printable, min_length: 1, max_length: 30)
+    |> StreamData.filter(&(!String.match?(&1, ~r/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/)))
```

Comment:
Note: this does not generate all invalid datetimes, as e.g. `"2024-99-99T99:99:99.371Z"` would match the regex. It's ok for now, though